tracing.go

```diff
@@ -185,7 +185,11 @@ func InjectOutboundSpan(response *OutboundCallResponse, headers map[string]strin
 		return headers // Tracer did not add any tracing headers, so return the original map
 	}
 	for k, v := range headers {
-		newHeaders[k] = v
+		// Some applications propagate all inbound application headers to outbound calls (issue #682).
+		// If those headers include tracing headers we want to make sure to keep the new tracing headers.
+		if _, ok := newHeaders[k]; !ok {
+			newHeaders[k] = v
```

Comment:
Do you know why we didn't get coverage from the added test here?

Reply:
Don't know. When I run this locally it shows the line as covered:
```
$ go test -v -run TestTracingSpanAttributes -coverprofile cover.out
=== RUN   TestTracingSpanAttributes
--- PASS: TestTracingSpanAttributes (0.00s)
PASS
coverage: 48.8% of statements
ok  	github.com/uber/tchannel-go	0.042s

$ go tool cover -html=cover.out
```

![image](https://user-images.githubusercontent.com/3523016/36509634-4cd47d8c-172e-11e8-922c-463d1845b6b9.png)

I added another check to ensure that non-tracing headers are propagated.